core/test/dataset_test.cpp

```diff
@@ -165,6 +165,20 @@ TEST(DatasetTest, setData_with_and_without_variances) {
   ASSERT_EQ(d.size(), 2);
 }
 
+TEST(DatasetTest, setData_updatyes_dimensions_correctly) {
+  /* See https://github.com/scipp/scipp/issues/819 */
```

Comment:
Maybe better explain the issue in a code comment? References to solved issues are maybe not that useful?

Reply:
I think the test itself explains the issue and the commit message explains the fix, to be honest I'm not sure  why I bothered with the link.